task/upload.go

```diff
@@ -57,6 +58,7 @@ type handleUploadVODParams struct {
 	profiles                 []api.Profile
 	targetSegmentSizeSecs    int64
 	catalystPipelineStrategy pipeline.Strategy
+	clipStrategy             video.ClipStrategy
```

Comment:
Doesn't seem to be a need for using the `catalyst-api` type here, since we are redefining it on `clients` and have to convert to that one in the end. So I'd say this should be just `*clients.ClipStrategy` (a pointer) and you can skip a couple of object recreations as well.